publication-commons/src/main/java/no/unit/nva/service/impl/DynamoDBPublicationService.java

```diff
@@ -0,0 +1,120 @@
+package no.unit.nva.service.impl;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.document.DynamoDB;
+import com.amazonaws.services.dynamodbv2.document.Index;
+import com.amazonaws.services.dynamodbv2.document.Item;
+import com.amazonaws.services.dynamodbv2.document.ItemCollection;
+import com.amazonaws.services.dynamodbv2.document.QueryOutcome;
+import com.amazonaws.services.dynamodbv2.document.Table;
+import com.amazonaws.services.dynamodbv2.document.spec.QuerySpec;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import no.unit.nva.Environment;
+import no.unit.nva.Logger;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.PublicationSummary;
+import no.unit.nva.service.PublicationService;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+
+import static no.unit.nva.PublicationHandler.ENVIRONMENT_VARIABLE_NOT_SET;
+
+public class DynamoDBPublicationService implements PublicationService {
+
+    public static final String NOT_IMPLEMENTED = "Not implemented";
+    public static final String TABLE_NAME_ENV = "TABLE_NAME";
+    public static final String BY_PUBLISHER_INDEX_NAME_ENV = "BY_PUBLISHER_INDEX_NAME";
+    public static final String DYNAMODB_KEY_DELIMITER = "#";
+
+    private final Index byPublisherIndex;
+    private final ObjectMapper objectMapper;
+
+    /**
+     * Constructor for DynamoDBPublicationService.
+     *
+     */
+    public DynamoDBPublicationService(ObjectMapper objectMapper, Index byPublisherIndex) {
+        this.objectMapper = objectMapper;
+        this.byPublisherIndex = byPublisherIndex;
+    }
+
+    /**
+     * Constructor for DynamoDBPublicationService.
+     *
+     */
+    public DynamoDBPublicationService(AmazonDynamoDB client, ObjectMapper objectMapper, Environment environment) {
+        String tableName = environment.get(TABLE_NAME_ENV).orElseThrow(
```

Comment:
throw the Exception in the Environment. We almost always throw exception. You can keep a getOptional in case we do not want to throw an exception

Reply:
You are already doing this in the upcoming common lib right?

Reply:
yes